### PR TITLE
fix: [CDS-54130]: bring back project error placement into dropdown, add validation for trafficPercent field

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary

- Bring Project field error message back into dropdown as it can be very large at times and takes up lot of space because of that.
- Add (optional) label to Field Mask field in steps
- Traffic Percent should not allow negative percentage (-25).

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger coverage

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/10-common/strings/strings.en.yaml [hunk 1]
```diff
@@ -1358,3 +1358,4 @@ tickets:
   tickets: 'Tickets'
   ticketComment: 'Append comment to ticket when Security Issue changes are detected (coming soon)'
   ticketExemption: 'Create a ticket when an Exemption is created (coming soon)'
+conNotBeNegativeNumber: '{{name}} can not be a negative number'
```
Comment (rameshx) [inline src/modules/10-common/strings/strings.en.yaml:1361]:
```suggestion
canNotBeNegativeNumber: '{{name}} can not be a negative number'
```

File: src/modules/75-cd/components/PipelineSteps/GoogleCloudFunction/TrafficShiftExecutionStepEdit.tsx [hunk 2]
```diff
@@ -62,7 +62,19 @@ const TrafficShiftExecutionStepEdit = (
...
+            getString('validation.timeout10SecMinimum')
+          ),
+          spec: Yup.object().shape({
+            trafficPercent: Yup.number()
+              .min(0)
+              .max(100)
+              .typeError(
+                getString('common.validation.fieldIsRequired', {
+                  name: getString('cd.steps.googleCloudFunctionCommon.trafficPercent')
+                })
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/GoogleCloudFunction/TrafficShiftExecutionStepEdit.tsx:75]: If `trafficPercent` is both required and can not be negative, I think we need two different error messages here.

Comment (rameshx) [reply src/modules/75-cd/components/PipelineSteps/GoogleCloudFunction/TrafficShiftExecutionStepEdit.tsx:75]: maybe the newly added string `canNotBeNegativeNumber` needs to be used here

Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/GoogleCloudFunction/TrafficShiftExecutionStepEdit.tsx:75]: min and max take care of negative number not allowed use case. typeError kicks in as soon as empty value is given. 
I tried just required and it did not work, so had to use typeError.

**Review by rameshx — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**